frontend/lib/src/components/widgets/DataFrame/columns/NumberColumn.ts

```diff
@@ -58,28 +60,24 @@ export interface NumberColumnParams {
  * This supports float, integer, and unsigned integer types.
  */
 function NumberColumn(props: BaseColumnProps): BaseColumn {
-  const arrowTypeName = getTypeName(props.arrowType)
-  let format = undefined
-  if (arrowTypeName === "timedelta64[ns]") {
-    // Use duration formatting for timedelta64[ns] type:
-    format = "duration[ns]"
-  } else if (arrowTypeName.startsWith("period[")) {
-    // Use period formatting for period types:
-    format = arrowTypeName
-  }
   const parameters = mergeColumnParameters(
     // Default parameters:
     {
       // Set step to 1 for integer types
       step: isIntegerType(props.arrowType) ? 1 : undefined,
       // if uint (unsigned int), only positive numbers are allowed
       min_value: isUnsignedIntegerType(props.arrowType) ? 0 : undefined,
-      format,
     } as NumberColumnParams,
     // User parameters:
     props.columnTypeOptions
   ) as NumberColumnParams
 
+  // If not custom format is provided & the column type is duration or period,
+  // instruct the column to use the arrow formatting for the display value.
+  const useArrowFormatting =
```

Comment:
nit: it looks like this variable is first used in line 195. I would move it directly before that within the `try` block

Reply:
ahhh the code below is in a callback?! I did not realize, thanks for pointing that out. Makes a lot of sense 👍